src/OrchardCore.Modules/OrchardCore.Navigation/NavigationShapes.cs

```diff
@@ -38,34 +38,37 @@ public void Discover(ShapeTableBuilder builder)
 
                     var viewContextAccessor = context.ServiceProvider.GetRequiredService<ViewContextAccessor>();
                     var viewContext = viewContextAccessor.ViewContext;
-
-                    var navigationManager = context.ServiceProvider.GetRequiredService<INavigationManager>();
+                    var navigationManagers = context.ServiceProvider.GetServices<INavigationManager>();
                     var shapeFactory = context.ServiceProvider.GetRequiredService<IShapeFactory>();
                     var httpContextAccessor = context.ServiceProvider.GetRequiredService<IHttpContextAccessor>();
-                    var menuItems = await navigationManager.BuildMenuAsync(menuName, viewContext);
-                    var httpContext = httpContextAccessor.HttpContext;
 
-                    if (httpContext != null)
+                    foreach (var navigationManager in navigationManagers)
                     {
-                        // adding query string parameters
-                        var route = menu.GetProperty<RouteData>("RouteData");
-                        var routeData = new RouteValueDictionary(route.Values);
-                        var query = httpContext.Request.Query;
+                        var menuItems = await navigationManager.BuildMenuAsync(menuName, viewContext);
+                        var httpContext = httpContextAccessor.HttpContext;
 
-                        if (query != null)
+                        if (httpContext != null)
                         {
-                            foreach (var pair in query)
+                            // adding query string parameters
```

Comment:
Can we extract this into a `private` method?

Reply:
There will be no reusability, but I see your point. I don't think it is necessary here.